client/components/domains/register-domain-step/index.jsx

```diff
@@ -1240,7 +1237,6 @@ class RegisterDomainStep extends React.Component {
 
 		const shouldHideFreeDomainExplainer =
 			this.props.selectedFreePlanInSwapFlow || this.props.selectedPaidPlanInSwapFlow;
```

Comment:
This currently shows the explainer text in the /personal, /premium, /business, /ecommerce, and /domain flows. It wouldn't make sense there because there is no plans step in those flows. We should add a prop [in this file](https://github.com/Automattic/wp-calypso/blob/master/client/signup/config/steps-pure.js) like `showFreeDomainExplainer` and explicitly set it to false in all these flows.